src/errors.rs

```diff
@@ -1,35 +1,43 @@
 use hyper;
 use std::convert::From;
-use serde_json::Error as JsonError;
+use serde_json;
 use std::num::ParseIntError;
 
-/// Handy type alias for a `Result` with `HueError`
-pub type Result<T> = ::std::result::Result<T, HueError>;
+impl From<::json::Error> for HueError {
+    fn from(e: ::json::Error) -> HueError {
+        HueErrorKind::BridgeError(e.address, e.description, From::from(e.code)).into()
+    }
+}
 
-#[derive(Debug)]
-/// Wrapper for all errors that can occur in this crate
-pub enum HueError {
-    /// The response from the bridge was malformed
-    ///
-    /// This doesn't happen in practice
-    MalformedResponse,
-    /// An error that occured in the bridge
-    BridgeError{
-        /// The URI the error happened on
-        address: String,
-        /// A short description of the error
-        description: String,
-        /// The `BridgeError`
-        error: BridgeError
-    },
-    /// A `serde_json::error::Error`
-    JsonError(JsonError),
-    /// A `hyper::Error`
-    HyperError(hyper::Error),
-    /// An `std::num::ParseIntError`
-    ParseIntError(ParseIntError)
+error_chain! {
+    
+    types {
+        HueError, HueErrorKind, ResultExt;
```

Comment:
You should add the Result as the fourth field here, so the macro also can generate that.